### DISCUSSION TITLE
Official `gatsby-source-bigcommerce` plugin

### DISCUSSION THREAD

**Discussion by chanceaclark** [gatsbyjs/gatsby#37446] [Ideas / Feature Requests]
👋  BigCommerce is looking to build out an official `gatsby-source-bigcommerce` plugin to ship within the Gatsby monorepo. We wanted to reach out about getting support for this official plugin from the Gatsby team as we would be building out the core foundations and initial data sourcing. Following the source plugin, we also wish to ship a starter repo but we can push that conversation to another discussion.

## Background research

Currently, there are three unofficial plugins:
- ⚠️  https://github.com/thirdandgrove/gatsby-source-bigcommerce
- ⚠️  https://github.com/Epic-Design-Labs/gatsby-source-bigcommerce
- ☑️ https://github.com/guyromellemagayano/gatsby-source-bigcommerce-v2/blob/main/src/gatsby-node.js

The `thirdandgrove` plugin has been unsupported for a while and have seen a few forks from that project; the last commit, as of writing this, was 3 years ago. We, at BigCommerce, have reached out to them to see if they would transfer the `gatsby-source-bigcommerce` npm package over to us. When we get ownership, we can figure it out from there.

~The `Epic-Design-Labs` plugin is the only one that is actively being supported. Since we don't want to step on anyone's toes we'd want to reach out to their team and work with them to ensure we have feature parity, to an extent:tm:.~ Edit: The `Epic-Design-Labs` plugin was deprecated and ownership was transferred to the `gatsby-source-bigcommerce-v2` plugin.

## Core functionality

After doing some of our own research and prototyping, here are some of the core the features/functionality ship in addition to feature parity of unofficial plugins:

### Multi-storefront

BigCommerce now has first-class support for multiple storefronts per BigCommerce account. We want to utilize this capability to support headless sites, specifically Gatsby in this case.

### Feature parity

As seen from `Epic-Design-Labs` and `thridandgrove`, here are some of the features that we will want to port over. Let me know if I'm missing some as this is pretty shallow. Some of the parity features might not be relevant with GraphQL.

- Gatsby preview
- Caching
- Additional headers

**Comment by mguyromelle** [comment]
Hi @chanceaclark I would like to help with this initiative. I am currently the owner of this unofficial `gatsby` plugin, https://www.npmjs.com/package/gatsby-source-bigcommerce-v2. Let's connect when you have the time. I will update the plugin with your help 😉  Cheers 🍻
